tensorflow/compiler/xla/service/gpu/ir_emitter.cc

```diff
@@ -315,7 +350,9 @@ bool IrEmitter::MaybeEmitDirectAtomicOperation(
                       ? llvm::AtomicRMWInst::Min
                       : llvm::AtomicRMWInst::UMin;
     AtomicRMW(opcode, output_address, source, llvm::MaybeAlign(),
-              llvm::AtomicOrdering::SequentiallyConsistent);
+              llvm::AtomicOrdering::SequentiallyConsistent,
+              (isAMDGPU) ? b_.getContext().getOrInsertSyncScopeID("agent") :
```

Comment:
This ternary is used quite a few times, factor out into a function?

Reply:
done.